.dotnet/README.md

```diff
@@ -798,6 +799,73 @@ Console.WriteLine($"[ASSISTANT]: {message}");
+//       ChatClient.CompleteChat(messages, options), will still use input information provided at the client
+//       or method level. In this case, any value provided via JSON or a dynamic object for "model" or
+//       "messages" will be overwritten by the values provided to the model and CompleteChat() method,
+//       respectively.
+ChatClient client = new("gpt-4o-mini", Environment.GetEnvironmentVariable("OPENAI_API_KEY"));
+ChatCompletion completion = client.CompleteChat(["Say hello. In JSON format, please!"], chatOptions);
+
+// Response types can also be serialized to BinaryData via ModelReaderWriter.Write(). This BinaryData can, in turn,
+// be used with typical System.Text.Json APIs.
+BinaryData completionBytes = ModelReaderWriter.Write(completion);
+JsonNode completionJsonNode = JsonNode.Parse(completionBytes);
+
+string lowLevelRetrievedRole = completionJsonNode["choices"][0]["message"]["role"].GetValue<string>();
+string lowLevelRetrievedContent = completionJsonNode["choices"][0]["message"]["content"].GetValue<string>();
+Console.WriteLine($"[{lowLevelRetrievedRole}]: {lowLevelRetrievedContent}");
+```
+
+Important things to note about direct serialization with `ModelReaderWriter`:
+
+- By design, library convenience types do *not* always represent a 1-to-1 mapping of a JSON schema component, as the library endeavors to improve some dimensions of idiomatic usability relative to a direct projection.
+- Deserialization of library convenience types may sometimes require providing low-level JSON that is required by the underlying JSON schema, even in cases where it is later modified when using convenience methods. This is shown above with how `model` and `messages` JSON properties are provided as part of a serialized `ChatCompletionOptions` basis even though those values are later replaced by the `model` provided to `ChatClient` and the `messages` provided to `CompleteChat(messages, options)`. Attempting to deserialize an instance of `ChatCompletionsOptions` *without* these values will result in an exception related to the missing required schema component.
+- Some nested library types are not directly serializable or deserializable with `ModelReaderWriter` because they don't map to a direct JSON schema component equivalent. In these cases, if `ModelReaderWriter` use is desired, work from a parent type so that manual serialization and deserialization can appropriately handle the library-only types. As an example of this, `ModerationClient` can request `ModerationResult` instances that expose a number of `ModerationCategory` properties; `ModerationCategory` reassembles information from multiple locations in the raw response JSON and thus can't be used with `ModelReaderWriter`. Instead, the parent `ModerationResult` can be directly interacted with using `ModelReaderWriter`, which will automatically serialize or deserialize the inferred `ModerationCategory` properties from the parent data.
```

Comment:
I think it would be good to mention that this only works with classes the implement `IJsonModel`.